tests/esm-path-mapping/path-relative.ts

```diff
@@ -0,0 +1,2 @@
+// Should fail because path mapping should be ignored
+import foo from './1-foo';
```

Comment:
This test is failing because this import is causing a TypeScript type-checking error, so the path-mapping error never happens.  You can tackle this a few different ways:

You can avoid the typechecking error with `@ts-expect-error` or enable `transpileOnly`.

Alternatively, can we test when `./1-foo.ts` *does* exist?  I'm not sure, but that might work.  The test's goal is to prove that path mapping is not applied.

What if we have `paths: {"*": ["subdirectory/*"]}`
It'll try to map everything into that subdirectory.

If we have two files `subdirectory/1-foo.ts` and `1-foo.ts` and we do `import from './1-foo'`, this test needs to prove that we end up loading `1-foo.ts`, *not* subdirectory/1-foo.ts`

Reply:
I think we need to test both false positives & false negatives. If `@ts-expect-error` works that keeps the fix precise/localized.

Reply:
This path mapping pull request only affects node's resolution behavior; it does not affect the typechecker at all.  This means the code changes in this pull request can be fully tested with typechecking disabled: `--transpileOnly` mode.  How about we do that for simplicity?  For all of these test cases, we use `transpileOnly` mode.

Reply:
Happy with either! :)